src/GAP/iso_gap_oscar.jl

```diff
@@ -0,0 +1,192 @@
+# Basically the same as the usual preimage function but without a type check
+# since we don't have elem_type(C) in this case
+function preimage(M::Map{D, C}, a) where {D <: GapObj, C}
+  if isdefined(M.header, :preimage)
+    p = M.header.preimage(a)
+    return p
+  end
+  error("No preimage function known")
+end
+
+################################################################################
+#
+#  Ring isomorphism
+#
+################################################################################
+
+# Compute the isomorphism between the GAP domain `F`
+# and a corresponding Oscar object.
+function _iso_gap_oscar(F::GAP.GapObj)
+   if GAP.Globals.IsField(F)
+     if GAPWrap.IsFinite(F)
+       if GAP.Globals.IsPrimeField(F)
+         return _iso_gap_oscar_field_finite_prime(F)
+       else
+         return _iso_gap_oscar_field_finite_ext(F)
+       end
+     else
+       if GAP.Globals.IsRationals(F)
+         return _iso_gap_oscar_field_rationals(F)
+       elseif GAP.Globals.IsCyclotomicCollection(F)
+         if GAP.Globals.IsCyclotomicField(F)
+           return _iso_gap_oscar_field_cyclotomic(F)
+         end
+       end
+     end
+   end
+
+   error("no method found")
+end
+
+# For finite prime fields, choose `GF(p)`
+# (of type `Nemo.GaloisField` if `p` is small (fits into `UInt64`),
+# `Nemo.GaloisFmpzField` otherwise)).
+function _iso_gap_oscar_field_finite_prime(F::GAP.GapObj)
+   p = characteristic(F)  # of type `fmpz`
+   if p < fmpz(2)^64
+     p = UInt64(p)
+   end
+   G = GF(p)
+   e = GAPWrap.One(F)
+
+   f(x) = G(fmpz(GAPWrap.IntFFE(x)))
```

Comment:
A tiny optimization is possible:
```suggestion
   function f(x) y=GAPWrap.IntFFE(x); return G(y isa Int ? y : fmpz(y))
```
I actually ended up doing this in a bunch of places, and perhaps it'd be useful to have an abbreviation for this... E.g.
```julia
IntegerUnion(y) = fmpz(y)
IntegerUnion(y::Int) = y
```